File: client/android/app/src/main/java/org/cna/keyple/demo/sale/android/nfc/slave/di/UIModule.java
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 ********************************************************************************/
package org.cna.keyple.demo.sale.android.nfc.slave.di;

import org.cna.keyple.demo.sale.android.nfc.slave.di.scopes.ActivityScoped;
import org.cna.keyple.demo.sale.android.nfc.slave.di.scopes.FragmentScoped;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.activities.MainActivity;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.activities.SplashScreenActivity;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.CardReaderFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.CardSummaryFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.ChargeCardFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.ChargeDeviceFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.DeviceStatusFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.PaymentValidatedFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.SelectDeviceFragment;
import org.cna.keyple.demo.sale.android.nfc.slave.ui.fragments.SettingsFragment;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class UIModule {
    @ActivityScoped
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

    @ActivityScoped
    @ContributesAndroidInjector
    abstract SplashScreenActivity splashScreenActivity();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract SelectDeviceFragment selectDeviceFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract CardReaderFragment cardReaderFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract DeviceStatusFragment deviceStatusFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract CardSummaryFragment cardSummaryFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract PaymentValidatedFragment paymentValidatedFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract ChargeCardFragment chargeCardFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract ChargeDeviceFragment chargeDeviceFragment();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract SettingsFragment settingsFragment();
}

File: client/android/app/src/main/kotlin/org/eclipse/keyple/demo/remote/di/UIModule.kt
/*
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information
 * regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */


package org.eclipse.keyple.demo.remote.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import org.eclipse.keyple.demo.remote.di.scopes.ActivityScoped
import org.eclipse.keyple.demo.remote.ui.CardReaderActivity
import org.eclipse.keyple.demo.remote.ui.CardSummaryActivity
import org.eclipse.keyple.demo.remote.ui.ChargeActivity
import org.eclipse.keyple.demo.remote.ui.ChargeResultActivity
import org.eclipse.keyple.demo.remote.ui.CheckoutActivity
import org.eclipse.keyple.demo.remote.ui.ConfigurationSettingsActivity
import org.eclipse.keyple.demo.remote.ui.HomeActivity
import org.eclipse.keyple.demo.remote.ui.PaymentValidatedActivity
import org.eclipse.keyple.demo.remote.ui.SelectTicketsActivity
import org.eclipse.keyple.demo.remote.ui.ServerSettingsActivity
import org.eclipse.keyple.demo.remote.ui.SettingsMenuActivity
import org.eclipse.keyple.demo.remote.ui.SplashScreenActivity

@Module
abstract class UIModule {
    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun splashScreenActivity(): SplashScreenActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun homeActivity(): HomeActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun configurationSettingsActivity(): ConfigurationSettingsActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun serverSettingsActivity(): ServerSettingsActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun settingsMenuActivity(): SettingsMenuActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun cardReaderActivity(): CardReaderActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun cardSummaryActivity(): CardSummaryActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun selectTicketsActivity(): SelectTicketsActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun checkoutActivity(): CheckoutActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun paymentValidatedActivity(): PaymentValidatedActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun chargeCardActivity(): ChargeActivity?

    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun chargeResultActivity(): ChargeResultActivity?
}